File: 1598. Crawler Log Folder/minOperations.py
from typing import List

class Solution:
  def minOperations(self, logs: List[str]) -> int:
    minOpCnt = 0

    for op in logs:
      if op == '../':
        minOpCnt -= 1 if minOpCnt > 0 else 0
      elif op != './':
        minOpCnt += 1

    return minOpCnt
  

logs = ["d1/", "d2/", "../", "d21/", "./"]
solution = Solution()
print('minOperations: ', solution.minOperations(logs))

File: 1598. Crawler Log Folder/minOperations.js
/**
 * @param {string[]} logs
 * @return {number}
 */
const minOperations = (logs) => {
  let minOpCnt = 0;

  for (let i = 0; i < logs.length; i++) {
    if (logs[i] === "../") {
      minOpCnt -= minOpCnt > 0 ? 1 : 0;
    } else if (logs[i] !== "./") {
      minOpCnt++;
    }
  }

  return minOpCnt;
};

const logs = ["d1/", "d2/", "../", "d21/", "./"];
console.log("minOperations: ", minOperations(logs));
